balena/models/device_os.py

```diff
@@ -419,7 +430,7 @@ def __transform_host_apps(self, host_apps):
             os_versions_by_device_type[host_app_device_type].extend(self.__get_os_versions_from_releases(host_app['owns__release'], app_tags))
 
         for device_type in os_versions_by_device_type:
-            os_versions_by_device_type[device_type].sort(reverse=True, key=cmp_to_key(sort_version))
+            os_versions_by_device_type[device_type].sort(reverse=True, key=lambda x: x['formatted_version'])
```

Comment:
Is this making the sorting lexicographical (thus 2.89.0 would show before 2.9.0)?

Reply:
I've implemented a `compare` function based on `balena-semver.compare` and it is used to sort the versions here.